minio/commonconfig.py

```diff
@@ -417,9 +417,9 @@ def __init__(self, bucket_name, object_name, region=None, version_id=None,
 
     def _validate_size(self, object_size):
         """Validate object size with offset and length."""
-        def _get_error(name, value):
+        def make_error(name, value):
```

Comment:
This is what I would prefer and implement, and you are welcome to ignore this comment if you'd like;
`make_error` sounds like there is no error and you are making it in my "opinion".
We hit the error and we are forming/putting together a value error message to be raised. So, I would 'personally' call `make_error` method => `raise_value_error` and raise the `ValueError` inside the method, instead of returning the `ValueError` and then raise the returned value.

As I said, it is up to you, so no matter what it is a LGTM.

Reply:
I thought about it, but unfortunately there is no good way in Python to hint to the interpreter that the particular method is always going to throw an exception. Thus linters cannot properly predict the code flow after such method is called, because they cannot guess the correct flow. Example


```

def method_that_always_raises():
    raise ValueError


def some_other_method():
        method_that_always_raises()
        # here linters are lost and assume the second operator would still be executed
        assert False, 'This never happens'


def method_that_returns_exc():
    return ValueError


def some_other_method2():
        raise method_that_returns_exc()
        # here linters know for sure that method execution would never reach the next line
        assert False, 'This never happens'
```

Reply:
I see.
Thanks.